### PR TITLE
Add more content-view-related BATS tests

### PR DESCRIPTION
**Pull request by ianballou**
A recent Katello change I made introduced many changes to content-view related code for Pulp 3 Yum content.  I think this is as good a time as ever to make some more content-view focused BATS tests.

### CODE REVIEW

File: bats/fb-content-katello.bats [hunk 1]
```diff
@@ -284,3 +284,191 @@ setup() {
...
+}
+
+@test "create first component content view" {
+  hammer content-view create --organization="${ORGANIZATION}" \
+    --name="${CONTENT_VIEW_2}" | grep -q "Content view created"
+}
+
+@test "add yum and docker repos to first component content view" {
+  repo_id=$(hammer repository list --organization="${ORGANIZATION}" \
+    | grep ${YUM_REPOSITORY} | cut -d\| -f1 | egrep -i '[0-9]+')
```
Comment (ekohl) [inline bats/fb-content-katello.bats:311]: Does hammer have an output mode that's easier to parse?

File: bats/fb-content-katello.bats [hunk 2]
```diff
@@ -284,3 +284,191 @@ setup() {
...
+    --product="${PRODUCT}" --content-type="yum" --name "${YUM_REPOSITORY_2}" \
+    --url https://partha.fedorapeople.org/test-repos/separated/modules-rpms/ | grep -q "Repository created"
+  hammer repository synchronize --organization="${ORGANIZATION}" \
+    --product="${PRODUCT}" --name="${YUM_REPOSITORY_2}"
+}
+
+@test "create and sync rpm-deps repo" {
+  hammer repository create --organization="${ORGANIZATION}" \
+    --product="${PRODUCT}" --content-type="yum" --name "${YUM_REPOSITORY_3}" \
+    --url https://partha.fedorapeople.org/test-repos/separated/rpm-deps/ | grep -q "Repository created"
```
Comment (jlsherrill) [inline bats/fb-content-katello.bats:299]: would it be easy enough to change to use https://fixtures.pulpproject.org/  ?   i'd rather do that than a dev's fedorapeople page.  If they don't have the needed content to test this, than its fine

File: bats/fb-content-katello.bats (hunk 1 above)
Comment (jlsherrill) [reply bats/fb-content-katello.bats:311]: you might look into  `hammer --csv --no-headers`

File: bats/fb-content-katello.bats (hunk 2 above)
Comment (ekohl) [reply bats/fb-content-katello.bats:299]: And if they don't, are they interested in hosting them?

Comment (ehelms) [reply bats/fb-content-katello.bats:299]: That's an excellent site by them. If they don't have what's needed, ask them :) That site seems the perfect spot for all incantations of content repos needed.

File: bats/fb-content-katello.bats (hunk 1 above)
Comment (ehelms) [reply bats/fb-content-katello.bats:311]: I think you can have it spit it out as JSON.

File: bats/fb-content-katello.bats (hunk 2 above)
Comment (ianballou) [reply bats/fb-content-katello.bats:299]: I'll ask them if they have a repo pair they use for testing dep solving.  If not, maybe they can host this one.

File: bats/fb-content-katello.bats (hunk 1 above)
Comment (ianballou) [reply bats/fb-content-katello.bats]: `hammer --csv --no-headers` looked like it made for the easiest parsing.  Just had to tokenize by `,` and use the right column

File: bats/fb-content-katello.bats (hunk 2 above)
Comment (ianballou) [reply bats/fb-content-katello.bats:299]: Put in the request here: https://pulp.plan.io/issues/7333

**Comment by ehelms** [comment]
Generally LGTM for these additional tests if @ianballou and @jlsherrill are

**Comment by evgeni** [comment]
Is there a minimum Katello version required for these to work? Looking at the hammer commands my feeling is "no", but wanted to clarify as we're using the same tests also for downstream Satellite where 6.5 is still supported and that's Katello 3.10.

If the answer is yes, that shouldn't block this PR, just needs awareness for downstream :)

File: bats/fb-content-katello.bats (hunk 2 above)
Comment (jlsherrill) [reply bats/fb-content-katello.bats:299]: lets go ahead and merge this as is, it seems adding repos to the pulp fixtures is non-trivial

**Comment by jlsherrill** [comment]
all of these commands/features have been around for 5 years or more, so we should be okay.

**Comment by ianballou** [comment]
Looks like hammer hasn't supported the `--fields` option since 0.18.  I'm going to look into skipping those tests unless the hammer version is >= 0.18

File: bats/fb-content-katello.bats (hunk 2 above)
Comment (ekohl) [reply bats/fb-content-katello.bats:301]: Might be nice to add the link to that issue in a TODO inside the file somewhere.

Comment (ianballou) [reply bats/fb-content-katello.bats:303]: Added the TODO.

**Comment by ianballou** [comment]
I've tested this on a Sat 6.5 box and a Katello 3.16 box.  Both worked and ran the appropriate tests.

**Comment by ekohl** [comment]
@jlsherrill can you ACK? Not sure if you have permissions, but I can merge if you don't.

**Pull request merged**